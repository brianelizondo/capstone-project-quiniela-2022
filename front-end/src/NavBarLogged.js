import React from 'react';
import { useSelector } from 'react-redux';
import { NavDropdown } from 'react-bootstrap';

function NavBarLogged({ logout }) {
    const user = useSelector((state) => state.user);

    return (
        <>
        <NavDropdown.Item href={`/users/${user.username}/profile`}>Profile</NavDropdown.Item>
        <NavDropdown.Item href={`/users/${user.username}/quinielas/add`}>Add New Quiniela</NavDropdown.Item>
        <NavDropdown.Divider />
        <NavDropdown.Item onClick={logout}>Logout</NavDropdown.Item>
        </>
    );
}

export default NavBarLogged;