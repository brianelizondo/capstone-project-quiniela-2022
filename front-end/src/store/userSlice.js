import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
    name: 'user',
    initialState: {
        id: null, 
        firstName: null, 
        lastName: null, 
        email: null, 
        username: null, 
        isAdmin: false,
        token: null,
        isAuthenticated: false
    },
    reducers: {
        login: (state, action) => {
            state.id = action.payload.user.id;
            state.firstName = action.payload.user.firstName;
            state.lastName = action.payload.user.lastName;
            state.email = action.payload.user.email;
            state.username = action.payload.user.username;
            state.isAdmin = action.payload.user.isAdmin;
            state.token = action.payload.token;
            state.isAuthenticated = true;
        },
        update: (state, action) => {
            state = action.payload
        },
        logout: (state) => {
            state.id = null;
            state.firstName = null;
            state.lastName = null;
            state.email = null;
            state.username = null;
            state.isAdmin = null;
            state.token = null;
            state.isAuthenticated = false;
        }
    }
})
  
// Action creators are generated for each case reducer function
export const { login, update, logout } = userSlice.actions;
export default userSlice.reducer;