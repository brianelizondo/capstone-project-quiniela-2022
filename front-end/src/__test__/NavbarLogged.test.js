import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore } from 'redux-persist';
import store from '../store/store';
let persistor = persistStore(store);

import NavBarLogged from '../NavBarLogged';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'NavBarLogged' component without crashing", () => {
        render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <MemoryRouter>
                        <NavBarLogged />
                    </MemoryRouter>
                </PersistGate>
            </Provider>
        );
        const textFound = screen.getByText("Add New Quiniela");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'NavBarLogged' matches snapshot", () => {
        const {asFragment} = render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <MemoryRouter>
                        <NavBarLogged />
                    </MemoryRouter>
                </PersistGate>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
