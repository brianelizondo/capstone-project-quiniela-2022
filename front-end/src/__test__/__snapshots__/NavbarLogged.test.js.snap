// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'NavBarLogged' matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="dropdown-item"
    data-rr-ui-dropdown-item=""
    href="/users/null/profile"
  >
    Profile
  </a>
  <a
    class="dropdown-item"
    data-rr-ui-dropdown-item=""
    href="/users/null/quinielas/add"
  >
    Add New Quiniela
  </a>
  <hr
    class="dropdown-divider"
    role="separator"
  />
  <a
    class="dropdown-item"
    data-rr-ui-dropdown-item=""
    href="#"
    role="button"
    tabindex="0"
  >
    Logout
  </a>
</DocumentFragment>
`;
